### PR TITLE
Make dss and weighted_dss no-ops for empty fields.

### PR DESCRIPTION
**Pull request by sriharshakandala**
Make dss and weighted_dss no-ops for empty fields.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by sriharshakandala** [comment]
bors try

**Comment by sriharshakandala** [comment]
bors try

File: test/Spaces/ddss1.jl [hunk 1]
```diff
@@ -61,41 +71,54 @@ end
     @test Topologies.local_neighboring_elements(space.topology, 3) == [2, 4]
     @test Topologies.local_neighboring_elements(space.topology, 4) == [1, 3]
 
-    init_state(local_geometry, p) = (ρ = 1.0)
-    y0 = init_state.(Fields.local_geometry_field(space), Ref(nothing))
+    y0 = init_state_scalar.(Fields.local_geometry_field(space), Ref(nothing))
```
[the remark this answers is not in the document]
Comment (sriharshakandala) [reply test/Spaces/ddss1.jl:74]: Sure!

**Comment by sriharshakandala** [comment]
bors try

File: test/Spaces/ddss1.jl [hunk 2]
```diff
@@ -61,41 +71,54 @@ end
...
+@testset "test if dss is no-op on an empty field" begin
+    Nq = 3
+    space, comms_ctx = distributed_space((4, 1), (true, true), (Nq, 1, 1))
+    y0 = init_state_scalar.(Fields.local_geometry_field(space), Ref(nothing))
+
+    dims = (Nq, Nq, 0, 4)
+    array = similar(parent(y0), dims)
+    data = DataLayouts.rebuild(Fields.field_values(y0), array)
+    space = axes(y0)
+    empty_field = Fields.Field(data, space)
```
Comment (simonbyrne) [inline test/Spaces/ddss1.jl:104]: Should be easier to do
```suggestion
    empty_field = similar(y0, Tuple{})
```

**Review by simonbyrne — CHANGES_REQUESTED** [review]

**Comment by simonbyrne** [comment]
looks good other than the test change

**Pull request merged**